workers: fix typo (#16220)
cloudflare/cloudflare-docs: src/content/docs/workflows/build/trigger-workflows.mdx

## Workers API (Bindings)

You can interact with Workflows programmatically from any Worker script by creating a binding to a Workflow. A Worker can bind to multiple Workflows, including Workflows defined in other Workers projects (scripts) within your account.

You can interact with a Workflow:

* Directly over HTTP via the [`fetch`](/workers/runtime-apis/handlers/fetch/) handler
* From a [Queue consumer](/queues/configuration/javascript-apis/#consumer) inside a `queue` handler
* From a [Cron Trigger](/workers/configuration/cron-triggers/) inside a `scheduled` handler
* Within a [Durable Object](/durable-objects/).

:::note

New to Workflows? Start with the [Workflows tutorial](/workflows/get-started/guide/) to deploy your first Workflow and familiarize yourself with Workflows concepts.

:::

To bind to a Workflow from your Workers code, you need to define a [binding](/workers/wrangler/configuration/) to a specific Workflow. For example, to bind to the Workflow defined in the [get started guide](/workflows/get-started/guide/), you would configure a `wrangler.toml` with the below:

```toml title="wrangler.toml"
name = "workflows-tutorial"
main = "src/index.ts"

> compatibility_date = "2024-10-15"

Reviewer: Chnage to:
compatibility_date = "2024-10-22"

	},
};
```

### Inspect a Workflow's status

You can inspect the status of any running Workflow instance by calling `status` against a specific instance ID. This allows you to programmatically inspect whether an instance is queued (waiting to be scheduled), actively running, paused or errored.

```ts
let instance = await env.MY_WORKFLOW.get("abc-123")
let status = await instance.status() // Returns an InstanceStatus
```

The possible values of status are as follows:

```ts
  status:
    | "queued" // means that instance is waiting to be started (see concurrency limits)
    | "running"
    | "paused"
    | "errored"
    | "terminated" // user terminated the instance while it was running
    | "complete"
    | "waiting" // instance is hibernating and waiting for sleep or event to finish
    | "waitingForPause" // instance is finishing the current work to pause
    | "unknown";
  error?: string;
  output?: object;
};
```

### Explicitly pausing a Workflow

You can explicitly pause a Workflow instance (and later resume it) by calling `pause` against a specific instance ID.

```ts
let instance = await env.MY_WORKFLOW.get("abc-123")
await instance.pause() // Returns Promise<void>
```

> ### Resuming a Workflow

Reviewer 2: ```suggestion
### Resume a Workflow
```

You can resume a paused Workflow instance by calling `resume` against a specific instance ID.

```ts
let instance = await env.MY_WORKFLOW.get("abc-123")
await instance.resume() // Returns Promise<void>
```

Calling `resume` on an instance that is not currently paused will have no effect.

> ### Stopping a Workflow

Reviewer 2: ```suggestion
### Stop a Workflow
```

You can stop a Workflow instance by calling `abort` against a specific instance ID.

```ts
let instance = await env.MY_WORKFLOW.get("abc-123")
await instance.abort() // Returns Promise<void>
```

Once stopped, the Workflow instance *cannot* be resumed.

> ### Restarting a Workflow

Reviewer 2: ```suggestion
### Restart a Workflow
```

You can restart a Workflow instance by calling `restart` against a specific instance ID.

```ts
let instance = await env.MY_WORKFLOW.get("abc-123")

> await instance.restart() // Returns Promise<void>

Reviewer: Restarting a Workflow instance is not supported yet and will throw an Error. Add a warning about?

---
title: Trigger Workflows
pcx_content_type: concept
sidebar:
  order: 3

---

You can trigger Workflows both programmatically and via the Workflows APIs, including:

1. With [Workers](/workers) via HTTP requests in a `fetch` handler, or bindings from a `queue` or `scheduled` handler

> 2. Using the [Workflows REST API](/api/paths/accounts-account_id--workflows/get)

Reviewer 2: ```suggestion
2. Using the [Workflows REST API](/api/operations/wor-list-workflows)
```
Aligning the syntax of the REST API links so that the links are valid.

    | "terminated" // user terminated the instance while it was running
    | "complete"
    | "waiting" // instance is hibernating and waiting for sleep or event to finish
    | "waitingForPause" // instance is finishing the current work to pause
    | "unknown";
  error?: string;
  output?: object;
};
```

### Explicitly pause a Workflow

You can explicitly pause a Workflow instance (and later resume it) by calling `pause` against a specific instance ID.

```ts
let instance = await env.MY_WORKFLOW.get("abc-123")
await instance.pause() // Returns Promise<void>
```

### Resume a Workflow

You can resume a paused Workflow instance by calling `resume` against a specific instance ID.

```ts
let instance = await env.MY_WORKFLOW.get("abc-123")
await instance.resume() // Returns Promise<void>
```

Calling `resume` on an instance that is not currently paused will have no effect.

### Stop a Workflow

You can stop a Workflow instance by calling `abort` against a specific instance ID.

```ts
let instance = await env.MY_WORKFLOW.get("abc-123")
await instance.abort() // Returns Promise<void>
```

Once stopped, the Workflow instance *cannot* be resumed.

### Restart a Workflow

:::caution

> **Known issue**: Restarting a Workflow via the `restart()` method is not currently supported ad will throw an exception (error).

Reviewer 2: ```suggestion
**Known issue**: Restarting a Workflow via the `restart()` method is not currently supported and will throw an exception (error).
```

:::

```ts
let instance = await env.MY_WORKFLOW.get("abc-123")
await instance.restart() // Returns Promise<void>
```

Restarting an instance will immediately cancel any in-progress steps, erase any intermediate state, and treat the Workflow as if it was run for the first time.

## REST API (HTTP)

> Refer to the [Workflows REST API documentation](/api/paths/accounts-account_id--workflows--workflow_name--instances/post).

Reviewer 2: ```suggestion
Refer to the [Workflows REST API documentation](/api/operations/wor-create-new-workflow-instance).
```
